wgsl/index.bs

```diff
@@ -501,18 +520,49 @@ See [[#keyword-summary]] for the list of WGSL keywords.
 An <dfn>identifier</dfn> is a kind of [=token=] used as a name.
 See [[#declaration-and-scope]] and [[#directives]].
 
-The form of an identifier is defined via pattern-matching, except that
-an identifier:
-* must not have the same spelling as a [=keyword=] or as a [=reserved word=].
-* must not be `_` (a single underscore)
-* must not start with two underscores.
+The form of an identifier is based on the
+[=Unicode Standard Annex #31 for Unicode Version 14.0.0|Unicode Standard Annex #31=] for
+[[!UnicodeVersion14|Unicode Version 14.0.0]],
+with the following elaborations. Identifiers do not compare under canonical equivalence
+and they are not normalized otherwise,
+meaning that single accented code points are distinct from the same characters constructed
+by an accent and a letter, and Unicode code points that look similar are not
+treated as the same and they compare by their code points.
+
+Identifiers use the following profile described in terms of [=UAX31 Grammar=]:
+
+```
+<Identifier> := <Start> <Continue>* (<Medial> <Continue>+)*
+
+<Start> := XID_Start + U+005F
+<Continue> := <Start> + XID_Continue
+<Medial> :=
+```
+
+This means identifiers with non-ASCII characters like these are
+valid: `Δέλτα`, `réflexion`, `Кызыл`, `𐰓𐰏𐰇`, `朝焼け`, `سلام`, `검정`, `שָׁלוֹם`, `गुलाबी`, `փիրուզ`.
+
+With the following exceptions:
+* An identifier must not have the same spelling as a [=keyword=] or as a [=reserved word=].
+* An identifier must not be `_` (a single underscore)
+* An identifier must not start with two underscores.
 
 <div class='syntax' noexport='true'>
   <dfn for=syntax>ident</dfn> :
 
-    | `/([a-zA-Z_][0-9a-zA-Z][0-9a-zA-Z_]*)|([a-zA-Z][0-9a-zA-Z_]*)/`
+    | `/([\p{XID_Start}])|([_\p{XID_Start}][\p{XID_Continue}]+)/`
 </div>
 
+[=Unicode Character Database for Unicode Version 14.0.0=] includes non-normative listing
+with all valid characters
+of both [=UAX31 Lexical Classes|XID_Start=] and
+[=UAX31 Lexical Classes|XID_Continue=].
+
+Note: User agents should consider issuing developer-visible warnings in most or all
+situations where display of two identifiers with different code point sequences
```

Comment:
This is a little too broad, I think.
It should be ok to have different spellings of things that look the same, but for different declarations:

        fn here() {  let foo: i32 = 12; }
        fn there() { let foo: i32 = 14; ///  assume I spelled foo with a different codepoint sequence }

There is no problem to be flagged here, since the two scopes don't overlap.

I suggest qualifying it a little more tightly:

> Note:  A user agent or should issue developer-visible warnings when the meaning of a WGSL program would change if all instances of an identifier spelled with a specific codepoint sequence are replaced by another codepoint sequence that would appear the same to a reader.  For example, remapping each identifier to its canonical equivalent spelling under [=UAX15 Normalization Forms|Normaliztaion Form C (NFC)=] must permit the same set of declarations (avoiding new identifier collisions), and must not change how identifier uses [=resolve=] to their declarations.

Reply:
@dneto0 this makes sense. Change done, thanks!